zbar/image.c

```diff
@@ -207,11 +207,27 @@ zbar_image_t *zbar_image_copy (const zbar_image_t *src)
     dst->datalen = src->datalen;
     dst->data = malloc(src->datalen);
     assert(dst->data);
+    dst->inverted = src->inverted;
     memcpy((void*)dst->data, src->data, src->datalen);
     dst->cleanup = zbar_image_free_data;
     return(dst);
 }
 
+ 
+void zbar_image_invert(const zbar_image_t *img)
```

Comment:
You are modifying an object from a `const` pointer. `img` is supposed to be read-only.

Another issue is the modification of the original image. What if you want to support mirrored or rotated barcodes later?

Reply:
Well, it is true that I worked around the const stance. Semantically it makes sense to make all zbar_image_t const, but where should an inversion take place?
Reworking the qr decoding does not look to promising as the code is quite convoluted and implementing the same thing over and over only makes testing worse.